cmd/kops/create_cluster.go

```diff
@@ -181,6 +186,10 @@ func NewCmdCreateCluster(f *util.Factory, out io.Writer) *cobra.Command {
 	// Allow custom tags from the CLI
 	cmd.Flags().StringVar(&options.CloudLabels, "cloud-labels", options.CloudLabels, "A list of KV pairs used to tag all instance groups in AWS (eg \"Owner=John Doe,Team=Some Team\").")
 
+	// vSphere flags
+	cmd.Flags().StringVar(&options.VSphereServer, "vsphere-server", options.VSphereServer, "vsphere-server is required for vSphere. Set vCenter URL Ex: 10.192.10.30 or myvcenter.io (without https://)")
+	cmd.Flags().StringVar(&options.VSphereDatacenter, "vsphere-datacenter", options.VSphereDatacenter, "vsphere-datacenter is required for vSphere. Set the name of the datacenter in which to deploy Kubernetes VMs.")
+	cmd.Flags().StringVar(&options.VSphereResourcePool, "vsphere-resource-pool", options.VSphereDatacenter, "vsphere-resource-pool is required for vSphere. Set a valid Cluster, Host or Resource Pool in which to deploy Kubernetes VMs.")
```

Comment:
>> Reason being, flags and spec fields can be logged and will help in debugging
We can log environment variables too.

>> Users will end up setting too many env variables if all of these flags are converted to env variables.
Its the same argument for flags too. So, I dont see how one is superior over another in this respect.

Regarding Miao's comments, I dont see how flags are better. Kops can still stop by validating the environment variables at the beginning. If we have more then having flags is as complex/simple as environment variables.

I dont mind using flags here but I just dont see how it s better over environment variables. In fact from security perspective, it is better to have them as environment variables as nobody else expect the one using kops will know the vsphere related parameters as they dont appear in the OS process list.